hack/test-end-to-end-docker.sh

```diff
@@ -66,11 +76,16 @@ function cleanup()
     echo "[INFO] Stopping k8s docker containers"; docker ps | awk 'index($NF,"k8s_")==1 { print $1 }' | xargs -l -r docker stop
     if [[ -z "${SKIP_IMAGE_CLEANUP-}" ]]; then
       echo "[INFO] Removing k8s docker containers"; docker ps -a | awk 'index($NF,"k8s_")==1 { print $1 }' | xargs -l -r docker rm
+
+      # clean up any residual volume mounts
+      mount | grep pods | grep "${VOLUME_DIR}" | awk '{print $3}' | sudo xargs -r -n 1 umount
```

Comment:
>cleanup can't really make assumptions about the API being available... it gets called in response to an exit signal, which could be from a failure halfway through...
I thought we were talking about waiting for a condition after `delete all`, not verifying stuff following signal handling... did I misunderstand?